### PR TITLE
feat: add colemak/colemak-dh support

### LINKED ISSUE #69: Add Colemak layout support
Hello, I love Colemak and would love to contribute to the layouts, there are few common variants of Colemak. #45

### PR DESCRIPTION
**Pull request by delinx**
Closes #69

* Added Colemak layout
* Added Colemak-DH layout
* Added ASCII diagrams to all layouts for visual reference

### CODE REVIEW

**Comment by willothy** [comment]
This is awesome, thank you!

**Comment by tris203** [comment]
This is amazing @delinx. Thanks for your contribution!

Just for clean up I want to move the keys type annotation closer to the setting, in the keyboard,lua file. Then we need to add the new types to the type annotation.

Either you can do that if you're comfortable to do so? Or I can add to this PR later this evening

File: lua/hawtkeys/keyboards.lua [hunk 1]
```diff
@@ -1,5 +1,7 @@
 local M = {}
 -- TODO: Make this dynamic, loading from the keyboards directory
```
Comment (tris203) [inline lua/hawtkeys/keyboards.lua:2]: Move the HawtKeySupportKeyboardLayouts type definition to here
```lua
---@alias HawtKeySupportedKeyboardLayouts "qwerty" | "colmak" | "colemakdh"
```

File: lua/hawtkeys/keyboards/qwerty.lua [hunk 2]
```diff
@@ -35,3 +35,8 @@ local layout = {
 return {
     layout = layout,
 }
+
+------1-2-3-4-5-6-7-8-9-10
```
Comment (tris203) [inline lua/hawtkeys/keyboards/qwerty.lua]:
```suggestion
------1-2-3-4-5-6-7-8-9-0
```

File: lua/hawtkeys/keyboards/colemak-dh.lua [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+return {
+    layout = layout,
+}
+
+-- https://colemakmods.github.io/mod-dh/
+-- NOTE: Angled mode is not accounted for since that would involve moving punctuation
+
+------1-2-3-4-5-6-7-8-9-10
```
Comment (tris203) [inline lua/hawtkeys/keyboards/colemak-dh.lua]:
```suggestion
------1-2-3-4-5-6-7-8-9-0
```

File: lua/hawtkeys/keyboards/colemak.lua [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    [" "] = { finger = 4, row = 4, col = 6 }, -- Spacebar
+}
+
+return {
+    layout = layout,
+}
+
+-- https://colemak.com/
+
+------1-2-3-4-5-6-7-8-9-10
```
Comment (tris203) [inline lua/hawtkeys/keyboards/colemak.lua]:
```suggestion
------1-2-3-4-5-6-7-8-9-0
```

**Review by tris203 — CHANGES_REQUESTED** [review]

**Review by tris203 — APPROVED** [review]

**Comment by tris203** [comment]
LGTM. thanks for your contribution @delinx

**Pull request merged**